File: p10_multiplicar_matrices.f90
      PROGRAM MULTIPLICACION_MATRICES
      INTEGER R1, C1, R2, C2, I, J, K
      INTEGER A(100), B(100), C(100)
      PRINT *, 'Ingrese dimensiones de A (r1 c1):'
      READ *, R1, C1
      PRINT *, 'Ingrese dimensiones de B (r2 c2):'
      READ *, R2, C2

      IF (R1.LE.0 .OR. C1.LE.0 .OR. R2.LE.0 .OR. C2.LE.0) THEN
         PRINT *, 'Dimensiones deben ser positivas'
         STOP
      END IF
      IF (C1.NE.R2) THEN
         PRINT *, 'No se puede multiplicar: c1 debe ser igual a r2'
         STOP
      END IF


      DO I=1,R1
         DO J=1,C2
            C((I-1)*10+J) = 0
         END DO
      END DO

      PRINT *, 'Ingrese la matriz A:'
      DO I=1,R1
         DO J=1,C1
            READ *, A((I-1)*10+J)
         END DO
      END DO

      PRINT *, 'Ingrese la matriz B:'
      DO I=1,R2
         DO J=1,C2
            READ *, B((I-1)*10+J)
         END DO
      END DO

      DO I=1,R1
         DO J=1,C2
            DO K=1,C1
               C((I-1)*10+J) = C((I-1)*10+J) + A((I-1)*10+K) * B((K-1)*10+J)
            END DO
         END DO
      END DO

      PRINT *, 'Resultado C = A x B:'
      DO I=1,R1
         DO J=1,C2
            WRITE(*,999) C((I-1)*10+J)
         END DO
         PRINT *
      END DO

      PAUSE 'Presione ENTER para finalizar'

  999  FORMAT(I8)

      END

File: p10_multiplicar_matrices.pas
program MultiplicarMatrices;
uses crt;

const
  MAX = 20;  

var
  r1, c1, r2, c2, i, j, k: integer;
  A, B, C: array[1..MAX, 1..MAX] of integer;

begin
  clrscr;
  write('Ingrese dimensiones de A (r1 c1): ');
  readln(r1, c1);
  write('Ingrese dimensiones de B (r2 c2): ');
  readln(r2, c2);

  if (r1 <= 0) or (c1 <= 0) or (r2 <= 0) or (c2 <= 0) then
  begin
    writeln('Dimensiones deben ser positivas.');
    halt(0);
  end;

  if c1 <> r2 then
  begin
    writeln('No se puede multiplicar: c1 debe ser igual a r2.');
    halt(0);
  end;

  writeln('Ingrese la matriz A:');
  for i := 1 to r1 do
    for j := 1 to c1 do
    begin
      write('A[', i, ',', j, ']= ');
      readln(A[i, j]);
    end;

  writeln('Ingrese la matriz B:');
  for i := 1 to r2 do
    for j := 1 to c2 do
    begin
      write('B[', i, ',', j, ']= ');
      readln(B[i, j]);
    end;

  for i := 1 to r1 do
    for j := 1 to c2 do
      C[i, j] := 0;

  
  for i := 1 to r1 do
    for j := 1 to c2 do
      for k := 1 to c1 do
        C[i, j] := C[i, j] + A[i, k] * B[k, j];

  
  writeln;
  writeln('Resultado C = A x B:');
  for i := 1 to r1 do
  begin
    for j := 1 to c2 do
      write(C[i, j]:8);
    writeln;
  end;

  readln;
end.
